frontend/src/test/scala/bloop/dap/DebugServerSpec.scala

```diff
@@ -45,12 +45,82 @@ import com.microsoft.java.debug.core.protocol.Types
 import com.microsoft.java.debug.core.protocol.Types.SourceBreakpoint
 import monix.execution.Ack
 import monix.reactive.Observer
+import io.reactivex.Observable
 
 object DebugServerSpec extends DebugBspBaseSuite {
   private val ServerNotListening = new IllegalStateException("Server is not accepting connections")
   private val Success: ExitStatus = ExitStatus.Ok
   private val resolver = new BloopDebugToolsResolver(NoopLogger)
 
+  testTask("Performs Hot Code Replace", FiniteDuration(16, SECONDS)) {
+    val source2 =
+      """|/Main.scala
+         |class A {
+         |  def m() = {
+         |    println("B")
+         |  }
+         |}
+         |""".stripMargin
+    TestUtil.withinWorkspace { workspace =>
+      val source =
+        """|/Main.scala
+           |object Main {
+           |  def main(args: Array[String]): Unit = {
+           |    println("A")
+           |    new A().m()
+           |  }
+           |}
+           |""".stripMargin
+      val Asource =
+        """|/A.scala
+           |class A {
+           |  def m() = {
+           |    println("A")
+           |  }
+           |}
+           |""".stripMargin
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val projectDep = TestProject(workspace, "a", List(Asource))
+      val mainProject = TestProject(workspace, "r", List(source), List(projectDep))
+
+      loadBspStateWithTask(workspace, List(mainProject, projectDep), logger) { state =>
+        val testState = state.compile(mainProject).toTestState
+        val buildProject = testState.getProjectFor(mainProject)
+        def srcFor(srcName: String) =
+          buildProject.sources.map(_.resolve(s"$srcName")).find(_.exists).get
+        val `A.scala` = srcFor("../../a/src/A.scala")
+        val breakpoints = breakpointsArgs(srcFor("Main.scala"), 4)
```

Comment:
Also, if I tried to place a breakpoint in `A.scala` at line 3 (a `println` call), it would not be verified